src/Shared/UnitTests/TestEnvironment.cs

```diff
@@ -426,46 +436,72 @@ void AssertDictionaryInclusion(IDictionary superset, IDictionary subset, string
 
     public class BuildFailureLogInvariant : TestInvariant
     {
+        private const string MSBuildLogFiles = "MSBuild_*.txt";
         private readonly string[] _originalFiles;
 
         public BuildFailureLogInvariant()
         {
-            _originalFiles = Directory.GetFiles(Path.GetTempPath(), "MSBuild_*.txt");
+            _originalFiles = GetMSBuildLogFiles();
+        }
+
+        private string[] GetMSBuildLogFiles()
+        {
+            List<string> files = new();
+            string debugPath = FileUtilities.TempFileDirectory;
+            if (debugPath != null)
+            {
+                try
+                {
+                    files.AddRange(Directory.GetFiles(debugPath, MSBuildLogFiles));
+                }
+                catch (DirectoryNotFoundException)
+                {
+                    // Temp folder might have been deleted by other TestEnvironment logic
+                }
+            }
+
+            try
+            {
+                files.AddRange(Directory.GetFiles(Path.GetTempPath(), MSBuildLogFiles));
```

Comment:
Can `Path.GetTempPath()` be the same as `FileUtilities.TempFileDirectory` which would make this method return duplicates?

Reply:
At code level, It should not be the same.  I mean if they are it would be some error in code. 
But to make tests more stable I have changed it and returns `files.Distinct()` instead.